buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.java

```diff
@@ -101,55 +105,104 @@ private static void setupApiInformationAnalysis(Project project, LibraryExtensio
             });
-    project
-        .getTasks()
-        .register(
-            "apiInformation",
-            ApiInformationTask.class,
-            task -> {
-              task.setApiTxt(project.file("api.txt"));
-              task.setMetalavaJarPath(metalavaOutputJarFile.getAbsolutePath());
-              task.setSourceSet(mainSourceSet);
-              task.setOutputFile(outputFile);
-              task.setBaselineFile(project.file("baseline.txt"));
-              task.setOutputApiFile(outputApiFile);
-              if (project.hasProperty("updateBaseline")) {
-                task.setUpdateBaseline(true);
-              } else {
-                task.setUpdateBaseline(false);
-              }
-              task.dependsOn("getMetalavaJar");
-            });
-
-    project
-        .getTasks()
-        .register(
-            "generateApiTxtFile",
-            GenerateApiTxtFileTask.class,
-            task -> {
-              task.setApiTxt(project.file("api.txt"));
-              task.setMetalavaJarPath(metalavaOutputJarFile.getAbsolutePath());
-              task.setSourceSet(mainSourceSet);
-              task.setBaselineFile(project.file("baseline.txt"));
-              if (project.hasProperty("updateBaseline")) {
-                task.setUpdateBaseline(true);
-              } else {
-                task.setUpdateBaseline(false);
+    File apiTxt =
+        project.file("api.txt").exists()
+            ? project.file("api.txt")
+            : project.file(project.getRootDir() + "/empty-api.txt");
+    TaskProvider<ApiInformationTask> apiInfo =
+        project
+            .getTasks()
+            .register(
+                "apiInformation",
+                ApiInformationTask.class,
+                task -> {
+                  task.setApiTxt(apiTxt);
+                  task.setMetalavaJarPath(metalavaOutputJarFile.getAbsolutePath());
+                  task.setSourceSet(mainSourceSet);
+                  android
+                      .getLibraryVariants()
+                      .all(
+                          v -> {
+                            if (v.getName().equals("release")) {
+                              ArtifactView artifactView =
+                                  v.getCompileConfiguration()
+                                      .getIncoming()
+                                      .artifactView(
+                                          config ->
+                                              config.attributes(
+                                                  container ->
+                                                      container.attribute(
+                                                          Attribute.of(
+                                                              "artifactType", String.class),
+                                                          "jar")));
+                              task.setClassPath(artifactView.getArtifacts().getArtifactFiles());
```

Comment:
I tried to come up with a witty comment about how nested this is, but I couldn't :-) If you can come up with a way of reduce the nesting level a bit, it would be great, but I don't have a very strong opinion in this particular case.

Reply:
the nesting is gone now, it was unneeded code